### ISSUE TITLE
Map.rotate expands dimensions of a map for even modest roll angles

### ISSUE THREAD

**Issue by wtbarnes** [sunpy/sunpy#4234]
When calling `.rotate` on a `Map`, I've found recently that, for even very modest roll angles, the resulting map is expanded by a few pixels in both dimensions. For example,

```python
>>> import sunpy.map
>>> import sunpy.data.sample
>>> m = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
>>> m.dimensions
PixelPair(x=<Quantity 1024. pix>, y=<Quantity 1024. pix>)
>>> m_rot = m.rotate()
PixelPair(x=<Quantity 1028. pix>, y=<Quantity 1028. pix>)
```

In general, `.rotate` should of course change the dimensions so as to not crop or change the resolution of the image. However, whether this is the correct behavior or not, this marks an unexpected (or at the very least undocumented) change in behavior. For example, tests of the `register` (i.e. "aiaprep") function in aiapy [have begun failing on sunpy master](https://gitlab.com/LMSAL_HUB/aia_hub/aiapy/-/issues/69) because the dimensions of the level 1 and level 1.5 maps are no longer equivalent.

**Comment by ayshih** [comment]
Can you provide a better example of *changed* behavior?  `AIA_171_IMAGE` has a non-zero `CROTA2` angle (~arcmins), so the rotated dimensions has "always" been (1028, 1028).  For kicks, I've confirmed that that was true even in SunPy 0.9.

**Comment by ayshih** [comment]
This almost certainly is due to the recent fixes to `submap()` (#4134), and has nothing to do with rotation.

**Comment by wtbarnes** [comment]
The aiapy test failures do indeed seem to be due to the submap changes in #4134 which I embarrassingly did not pull into aiapy.

Closing as this seems to be a non-issue.
